sw/device/lib/drivers/power_manager/data/power_manager.c.tpl

```diff
@@ -0,0 +1,330 @@
+// Copyright 2022 OpenHW Group
+// Solderpad Hardware License, Version 2.1, see LICENSE.md for details.
+// SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
+
+#include "power_manager.h"
```

Comment:
We need to generate multiple pieces of code depending on the number of RAM blocks and use different MACRO inside. Maybe there is another way, I don't know.

Reply:
If the register map is regular, you could just specify the offset of the domain you want to power gate. However, this creates less explicit code. I don't have a strong opinion, except I prefer to avoid templates when they can be avoided. Also, your solution can create very large code if you have many power domain.

Reply:
Solved as you suggested, for the moment :)